File: php/ltx/src/InterplanetLTX/LtxSegmentTemplate.php
<?php
/**
 * LtxSegmentTemplate.php — Segment type + quantum entry
 * Story 33.4 — PHP LTX library
 */

namespace InterplanetLTX;

/** A segment type and quantum count template entry. */
readonly class LtxSegmentTemplate
{
    public function __construct(
        public string $type,
        public int    $q,
    ) {}
}

File: java/ltx/src/com/interplanet/ltx/LtxSegmentTemplate.java
package com.interplanet.ltx;

/**
 * LtxSegmentTemplate — a segment in a plan's segment list.
 * Has a type and a number of quanta.
 * Story 33.2 — Java LTX library
 */
public record LtxSegmentTemplate(
    /** Segment type, e.g. "TX", "RX", "PLAN_CONFIRM". */
    String type,
    /** Duration in quanta (minutes = q * quantum). */
    int q
) {}
